core/common/src/main/java/alluxio/exception/DirectoryNotInSyncException.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.exception;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Exception for situations where an operation fails because a directory is not in sync with the
+ * UFS. An example of this would be deleting a nonempty directory which has UFS contents not loaded
+ * into Alluxio.
+ */
+@ThreadSafe
+public class DirectoryNotInSyncException extends AlluxioException {
```

Comment:
@aaudiber part of the reason to introduce a new exception was so that it can be handled differently. in this case `IOException` could mean retry, whereas another exception implies that it will fail no matter how many times you try.

let me know what fits in well.